src/main/scala/uk/sky/fs2/kafka/topicloader/TopicLoader.scala

```diff
@@ -96,32 +107,41 @@ trait TopicLoader {
       consumer: KafkaConsumer[F, K, V]
   ): F[Option[NonEmptyMap[TopicPartition, LogOffsets]]] =
     for {
-      _                           <- consumer.subscribe(topics)
-      partitionInfo               <- topics.toList.flatTraverse(consumer.partitionsFor)
-      topicPartitions              = partitionInfo.map(pi => TopicPartition(pi.topic, pi.partition)).toSet
+      topicPartitions             <-
+        topics.toList
+          .flatTraverse(topic =>
+            // Assign doesn't support incremental subscription, so we must aggregate partitions per topic
+            consumer.assign(topic) *> partitionsForTopics(topics, consumer).map(_.toList)
+          )
+          .map(_.toSet)
       beginningOffsetPerPartition <- consumer.beginningOffsets(topicPartitions)
       endOffsets                  <- strategy match {
                                        case LoadAll       => consumer.endOffsets(topicPartitions)
                                        case LoadCommitted => earliestOffsets(consumer, beginningOffsetPerPartition)
                                      }
-      logOffsets                   = beginningOffsetPerPartition.map { case (partition, offset) =>
+      logOffsets                   = beginningOffsetPerPartition.map { (partition, offset) =>
                                        partition -> LogOffsets(offset, endOffsets(partition))
                                      }
       _                           <- consumer.unsubscribe
-    } yield {
-      val offsets = logOffsets.filter { case (_, o) => o.highest > o.lowest }
-      NonEmptyMap.fromMap(SortedMap.from(offsets))
-    }
+    } yield NonEmptyMap.fromMap(SortedMap.from(logOffsets))
```

Comment:
If we do the non-empty offset filtering here, then our consumer never subscribes to the empty partitions in loadAndRun.